bitbots_vision/bitbots_vision/vision_modules/yoeo_vision_components.py

```diff
@@ -0,0 +1,889 @@
+        self._debug_image.draw_field_boundary(convex_field_boundary_points, DebugImageColors.field_boundary_convex)
+
+    def set_image(self, image) -> None:
+        """
+        :param image: Image to run the component on
+        :type image:  numpy nd.array (height, width, channels)
+        """
+        self._field_boundary_detector.set_image(image)
+
+
+class YOEOLineDetectionComponent(IVisionComponent):
+    """
+    Component carries out the line detection using YOEO
+    """
+
+    def __init__(self, node: Node):
+        self._config: Dict = {}
+        self._debug_image: Optional[debug.DebugImage] = None
+        self._line_detector: Optional[yoeo_handler.IYOEOSegmentation] = None
+        self._node: Node = node
+        self._publisher: Optional[rclpy.publisher.Publisher] = None
+
+    def configure(self, config: Dict, yoeo: yoeo_handler.IYOEOHandler) -> None:
+        self._debug_image = DebugImageFactory.create(config)
+        self._line_detector = yoeo_handler.YOEOLineSegmentation(yoeo)
+
+        self._register_publisher(config)
+        self._config = config
+
+    def _register_publisher(self, new_config: Dict) -> None:
+        self._publisher = ros_utils.create_or_update_publisher(
+            self._node,
+            self._config,
+            new_config,
+            self._publisher,
+            'ROS_line_mask_msg_topic',
+            Image
+        )
+
+    def run(self, image_msg) -> None:
+        """
+        :param image_msg: Image message
+        :type image_msg:  sensor_msgs.msg._image.Image
+        """
+        line_mask = self._line_detector.get_mask_image()
+        line_mask_msg = self._create_line_mask_msg(image_msg, line_mask)
+        self._publish_line_mask_msg(line_mask_msg)
+
+        self._add_line_mask_to_debug_image(line_mask)
+
+    @staticmethod
+    def _create_line_mask_msg(image_msg, line_mask):
+        """
+        :param image_msg: Image message
+        :type image_msg:  sensor_msgs.msg._image.Image
+        :param line_mask: Line mask
+        :type line_mask: numpy ndarray (height, width, channels)
+        :rtype: sensor_msgs.msg._image.Image
+        """
+        return ros_utils.build_image_msg(image_msg.header, line_mask, '8UC1')
+
+    def _publish_line_mask_msg(self, line_mask_msg) -> None:
+        """
+        :param line_mask_msg: Line mask message
+        :type line_mask_msg: sensor_msgs.msg._image.Image
+        """
+        self._publisher.publish(line_mask_msg)
+
+    def _add_line_mask_to_debug_image(self, line_mask) -> None:
+        """
+        :param line_mask: Line mask
+        :type line_mask: numpy ndarray (height, width, channels)
+        """
+        self._debug_image.draw_mask(line_mask, color=DebugImageColors.lines)
+
+    def set_image(self, image) -> None:
+        """
+        :param image: Image to run the component on
+        :type image:  numpy nd.array (height, width, channels)
+        """
+        pass  # Intentional
```

Comment:
Why is the pass here intentional?

Reply:
I asked myself the same. Please add a reason.